src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -516,7 +521,7 @@ parsedQuery::Service Visitor::visit(Parser::ServiceGraphPatternContext* ctx) {
   std::vector<Variable> visibleVariablesSoFar = std::move(visibleVariables_);
   parsedQuery::GraphPattern graphPattern = visit(ctx->groupGraphPattern());
   std::vector<Variable> visibleVariablesServiceQuery =
-      ad_utility::removeDuplicates(std::move(visibleVariables_));
+      ad_utility::removeDuplicates(visibleVariables_);
```

Comment:
I see that since `removeDuplicates` takes a `const&` argument (it didn't always), the `std::move` doesn't make sense here.

But just to fully understand this, what happens if one uses `std::move(visibleVariables)` in this call? Is the effect the same as if the `std::move` wasn't there? I hope.

Reply:
Yes, the `std::move()` does nothing.
because `std::move(const T&)` technically returns a `const T&&` which is still `const` and doesn't allow to change the object.